subsys/net/ip/ipv4.c

```diff
@@ -141,41 +141,30 @@ enum net_verdict net_ipv4_process_pkt(struct net_pkt *pkt)
 		goto drop;
 	}
 
+	if (!net_ipv4_is_my_addr(&hdr->dst) &&
+	    !net_ipv4_is_addr_mcast(&hdr->dst) &&
+	    !(IS_ENABLED(CONFIG_NET_DHCPV4) && hdr->proto == IPPROTO_UDP &&
```

Comment:
And wait, we don't support reception of broadcast IPv4 packets? Because that's what I see here, and if I got it wrong, that's because this `if` is quite complicated.

Reply:
we only support it if dhcpv4 is enabled (and incoming packet is udp).

Reply:
> we only support it if dhcpv4 is enabled (and incoming packet is udp)

Weird. @jukkar: Is that known issue? That for sure would fail some tests in any decent IPv4 testsuite I guess.

Reply:
We support it if protocol is icmpv4 (see net_icmpv4_input()). Now I start to doubt my test...
So it should not be for us if proto is udp, dst is broadcast and dhcpv4 is disabled. I'll check

Reply:
@tbursztyka: Well, I see that your new code doesn't add or take anything which the old code had, so sorry for the side-talk here. I'm just surprised that folks like @rmstoi and @ozhuraki who work on testing didn't bring up IPv4 broadcast support issues.

Reply:
sure, anyway I'll update that patch to get a more readable test.